currency/manager.go

```diff
@@ -458,6 +458,40 @@ func (p *PairsManager) getPairStoreRequiresLock(a asset.Item) (*PairStore, error
 	return pairStore, nil
 }
 
+// UnmarshalJSON implements json.Unmarshal
+// Pairs.Unmarshal doesn't know what the delimiter is, so uses the first punctuation rune
+// This method uses the configFormat delimiter to improve that guess
+func (p *PairsManager) UnmarshalJSON(d []byte) error {
```

Comment:
Can strip the methods by converting to alias straight away so it doesn't recursively call. Then we don't need to assign. Maybe add a comment as to why. 🤷 
```diff
diff --git a/currency/manager.go b/currency/manager.go
index 304c9de03..cf98db0c5 100644
--- a/currency/manager.go
+++ b/currency/manager.go
@@ -462,12 +462,10 @@ func (p *PairsManager) getPairStoreRequiresLock(a asset.Item) (*PairStore, error
 // Pairs.Unmarshal doesn't know what the delimiter is, so uses the first punctuation rune
 // This method uses the configFormat delimiter to improve that guess
 func (p *PairsManager) UnmarshalJSON(d []byte) error {
-       type Alias PairsManager
-       var t Alias
-       if err := json.Unmarshal(d, &t); err != nil {
+       type alias *PairsManager
+       if err := json.Unmarshal(d, alias(p)); err != nil {
                return err
        }
-       *p = PairsManager(t) //nolint:govet // Mutex copying of lexical private alias is okay
        for _, s := range p.Pairs {
                cf := s.ConfigFormat
                if cf == nil {
```

Reply:
This is super cute; Thank you 🙏 
Pattern Squirrelled.

Fixed c6b57d02